### ISSUE TITLE
Error 500: SQLSTATE[HY000]: General error: 1100 Table 'xx' was not locked with LOCK TABLES

### ISSUE THREAD

**Issue by skylabz0rz** [espocrm/espocrm#548]
I am having this issue on multiple installs of Espo. It's random and I can't seem to figure out why or what is causing this issue.

This is for event entity's only it seems. I can't seem to get this error to duplicate on other entities. 

Here is the log output (some data hidden):

[2017-05-23 05:03:44] Espo.ERROR: API [POST]:/:controller, Params:Array (     [controller] => MaintenanceLog ) , InputData: {"status":"Held","dateStart":"2017-05-22 23:00:00","duration":300,"assignedUserId":"1","assignedUserName":"Adam","dateEnd":"2017-05-22 23:15:00","aircraftName":"Big Geek","aircraftId":"591f305a4ffbe1605","batteriesIds":["591f32628ec3f0a9e"],"batteriesNames":{"591f32628ec3f0a9e":"W19ACC26010616"},"description":"Battery discharge per the DJI Go app. Ran down to 8% and recharged.","flightLogsIds":[],"flightLogsNames":{},"reminders":[],"teamsIds":[],"teamsNames":{}} - SQLSTATE[HY000]: General error: 1100 Table 'maintenance_log' was not locked with LOCK TABLES [] []

[2017-05-23 05:03:44] Espo.ERROR: Display Error: SQLSTATE[HY000]: General error: 1100 Table 'maintenance_log' was not locked with LOCK TABLES, Code: HY000 URL: /uas/api/v1/MaintenanceLog [] []

The server specs:

- Intel(R) Xeon(R) CPU L5520 @ 2.27GHz (16 core)
- Memory: 74212152k/76546048k available (5461k kernel code, 1058328k absent, 1275568k reserved, 6958k data, 1316k init)
- CENTOS 6.8 x86_64
- PHP7
- MySQL 5.6.35

Any help would be appreciated. I can't seem to get around this.

Thanks!

**Comment by yurikuzn** [comment]
Can you try to add `'actionHistoryDisabled' => true` to your data/config.php and check. Please let us know whether the error still is occurring.

**Comment by yurikuzn** [comment]
I've just realised that this unlikely will help. I thought that we use LOCK TABLE in espocrm but seems we don't.

**Comment by skylabz0rz** [comment]
I am trying to troubleshoot this more. I have tried to repair the table in phpmyadmin, as well as analyze it. I also tried to force an entry in using phpmyadmin, which worked, but I still cannot create entries with Espo. I also have restarted mySQL and Apache just to double check that and none of which helped.

You were correct as I added the changed to /data/config.php and the error still remains.

**Comment by skylabz0rz** [comment]
So I figured out what was going on here. 

I changed the name of the entity to a number as I wanted the system to auto generate a name, which was a number with a specific prefix. In /custom/Espo/Custom/Resources/metadata/entityDefs I changed name to number. When I save it, cleared the cache and performed a rebuild, ill get the lock tables error. If I change it back to a varchar, the error goes away. Another entity is also structured like this, and it works. 

More digging to come.
